### PR TITLE
added lang sub-categories

### LINKED ISSUE #341: [Feature] Add sites and other resources for each technology
### Description

We can add websites for learning each technology like html, css etc. seperately in the resources section.

### Screenshots

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ujjawaltyagii**
### Added subcategories for languages like ``Go, JavaScript  and Python``. Also, added some resources in ``javascript`` category

- Made changes to ``database>index.ts`` , ``types>index.ts`` , ``data.ts`` and created files ``python.json`` , ``javascript.json`` , ``golang.json``

## Screenshots
image

image

## Note to reviewers
- You are free to add more resources in provided categories and if you want to add any other language in sub-category then feel free to do so or just msg me I will do it :)

### CODE REVIEW

**Comment by ujjawaltyagii** [comment]
@rupali-codes done !

**Review by rupali-codes — COMMENTED** [review]
I would like to suggest some changes, I guess we should create a new category for it, something like `documentory`, 'documentations' or `programming languages`, you can suggest names, and then add this resources in there

**Comment by ujjawaltyagii** [comment]
okay :)

**Comment by ujjawaltyagii** [comment]
@rupali-codes done with the changes !

**Review by rupali-codes — APPROVED** [review]
looks great to me! thanks a lot

File: database/languages/javascript.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        "name": "JavaScript Info",
+        "description": "It provides Modern JavaScript Tutorial with detailed explanations with examples and tasks.",
+        "url": "https://javascript.info/",
+        "category": "Other",
+        "subcategory": "javascript"
+    },
+    {
+        "name": "JavaScript-Tutorial",
+        "description": "It helps you learn the JavaScript programming language from scratch quickly and effectively.",
```
Comment (CBID2) [inline database/languages/javascript.json:11]:
```suggestion
        "description": "It helps you quickly and effectively learn JavaScript.",
```

File: database/languages/javascript.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[
+    {
+        "name": "JavaScript Info",
+        "description": "It provides Modern JavaScript Tutorial with detailed explanations with examples and tasks.",
```
Comment (CBID2) [inline database/languages/javascript.json:4]:
```suggestion
        "description": "It is a modern JavaScript tutorial that provides detailed explanation, examples, and tasks.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Great start! 😀 I just made some suggestions to clean up your descriptions a bit.

**Comment by ujjawaltyagii** [comment]
@CBID2 thanks for the feedback !

**Review by CBID2 — APPROVED** [review]
Well done! 😊 LGTM! 😁

**Pull request merged**